theano/gpuarray/basic_ops.py

```diff
@@ -1245,8 +1245,27 @@ class GpuJoin(HideC, Join):
 
     """
     _f16_ok = True
+    __props__ = ("view",)
     params_type = gpu_context_type
 
+    def __init__(self, view=-1):
+        self.view = view
+        if view != -1:
+            # since the first input is always the axis, the tensors
+            # start from index 1.
+            self.view_map = {0: [1 + view]}
+
+    # def __str__(self):
+    #     if self.view == -1:
+    #         return "Join"
+    #     else:
+    #         return super(Join, self).__str__()
+
+    # def __setstate__(self, d):
+    #     self.__dict__.update(d)
+    #     if not hasattr(self, "view"):
+    #         self.view = -1
```

Comment:
I removed ```__setstate__()``` but we can't remove  ```__str__()``` as  ```__props__```  has been defined and overrides the inherited  ```__str__()``` from "Join". Thus, I had to define it in GpuJoin as well. Also, I changed 

```python
return super(Join, self).__str__()
``` 
to 
```python
return "%s{%s}" % (
                self.__class__.__name__,
                ", ".join("%s=%r" % (p, getattr(self, p))
                          for p in self.__props__))
```

because this is what was being returned in ```__props__```.